Week3/homework/step2-6.js

```diff
@@ -4,13 +4,28 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const flattenedArr2d = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      flattenedArr2d.push(arr[i][j]);
+    }
+  }
+  return flattenedArr2d;
 }
 
 function flattenArray3d(arr) {
   // Replace this comment and the next line with your code
-  console.log(arr);
+  const flattenedArr3d = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      for (let k = 0; k < arr[i][j].length; k++) {
+        flattenedArr3d.push(arr[i][j][k]);
```

Comment:
Logic works fine but imagine if you have to do this for a 4-D or a 5-D array. What happens in that case? Suggestion: try reading about a programming concept called recursion and see if you can apply the same here. Also, look up for a new method in JavaScript named `Array.prototype.flat`.  🙂